docs/application/dotnet/guides/nui/texteditor.md

```diff
@@ -0,0 +1,113 @@
+
+The `TextEditor` also supports markup, and text can be scrolled vertically within it.
+
+**Figure: TextEditor**
+
+![TextEditor](./media/dali_texteditor.png)
+
+In this tutorial, the following subjects are covered:
+
+[TextEditor events](#1)<br>
+[Creating a TextEditor](#2)<br>
+[TextEditor Properties](#3)<br>
+
+<a name="1"></a>
+## TextEditor events
+
+The following table lists the basic signals provided by the `TextEditor` class.
+
+**Table: TextEditor input signals**
+
+| Input signal         | Description                              |
+| -------------------- | ---------------------------------------- |
+| `TextChanged`        | Emitted when the text changes.           |
+| `ScrollStateChanged` | Emitted when TextEditor scrolling is started or finished. |
+
+<a name="2"></a>
+## Creating a TextEditor
+
+The following basic example shows how to create a `TextEditor` object:
+
+```
+// Create a TextEditor instance
+Window window = Window.Instance;
+TextEditor editor = new TextEditor();
+editor.Position2D = new Position2D(10, 700);
+editor.Size2D = new Size2D(400, 90);
+editor.BackgroundColor = Color.Red;
+editor.PointSize = 20;
+editor.TextColor = Color.White;
+editor.Text = "This is a multiline text.\n I can write several lines.\n"
+window.Add(editor);
+```
+
+<a name="3"></a>
+## TextEditor Properties
+
+You can modify the `TextEditor` appearance and behavior through its properties.
+
+The following table lists the available `TextEditor` properties.
+
+**Table: TextEditor properties**
+
+| Property                           | Type        | Description                              |
+| ---------------------------------- | ----------- | ---------------------------------------- |
+| `Text`                             | String      | The text to display in UTF-8 format.      |
+| `TextColor`                        | Vector4     | The text color.                           |
+| `FontFamily`                       | String      | The requested font family.                |
+| `FontStyle`                        | PropertyMap | The requested font style.                 |
+| `PointSize`                        | Float       | The size of font in points.               |
+| `HorizontalAlignment`              | String      | The line horizontal alignment.            |
+| `ScrollThreshold`                  | Float       | Horizontal scrolling will occur if the cursor is this close to the control border. |
+| `ScrollSpeed`                      | Float       | The scroll speed in pixels per second.    |
+| `PrimaryCursorColor`               | Vector4     | The color to apply to the primary cursor. |
+| `SecondaryCursorColor`             | Vector4     | The color to apply to the secondary cursor. |
+| `EnableCursorBlink`                | Boolean     | Whether the cursor should blink or not.   |
```

Comment:
Suggestion : We are on text editor page now (and the table is about texteditor properties), so I think it would be ok not to emphasize `on text editor` again. What do you think? (Also, we have same issue above)